src/main/java/uk/gov/hmcts/reform/bulkscan/orchestrator/services/ccd/CcdCaseUpdater.java

```diff
@@ -0,0 +1,209 @@
+        }
+    }
+
+    @SuppressWarnings("squid:S2139") // exception handle + logging
+    private long updateCaseInCcd(
+        boolean ignoreWarnings,
+        String idamToken,
+        String s2sToken,
+        String userId,
+        ExceptionRecord exceptionRecord,
+        CaseUpdateDetails caseUpdateDetails,
+        CaseDetails existingCase
+    ) {
+        try {
+            StartEventResponse eventResponse = coreCaseDataApi.startForCaseworker(
+                idamToken,
+                s2sToken,
+                userId,
+                exceptionRecord.poBoxJurisdiction,
+                existingCase.getCaseTypeId(),
+                // when onboarding remind services to not configure about to submit callback for this event
+                caseUpdateDetails.eventId
+            );
+
+            return coreCaseDataApi.submitForCaseworker(
+                idamToken,
+                s2sToken,
+                userId,
+                exceptionRecord.poBoxJurisdiction,
+                existingCase.getCaseTypeId(),
+                ignoreWarnings,
+                CaseDataContent
+                    .builder()
+                    .caseReference(exceptionRecord.id)
+                    .data(caseUpdateDetails.caseData)
+                    .event(Event
+                        .builder()
+                        .id(eventResponse.getEventId())
+                        .summary("Case updated")
+                        .description(
+                            format(
+                                "Case with case ID %s updated based on exception record ref %s",
+                                existingCase.getId(),
+                                exceptionRecord.id
+                            )
+                        )
+                        .build()
+                    )
+                    .eventToken(eventResponse.getToken())
+                    .build()
+            ).getId();
+        } catch (FeignException exception) {
+            log.error(
+                "Failed to update case for {} jurisdiction with case ID {} from exception record {}. "
+                    + "Service response: {}",
+                exceptionRecord.poBoxJurisdiction,
+                existingCase.getId(),
+                exceptionRecord.id,
+                exception.contentUTF8(),
+                exception
+            );
+
+            throw exception;
+        } catch (Exception exception) {
+            log.error(
+                "Failed to update case for {} jurisdiction with case ID {} from exception record {}",
+                exceptionRecord.poBoxJurisdiction,
+                existingCase.getId(),
+                exceptionRecord.id,
+                exception
+            );
+
+            throw exception;
+        }
+    }
+
+    private void checkBulkScanReferenceIsSet(Map<String, ?> caseData, String exceptionRecordId) {
+        if (!caseData.containsKey(EXCEPTION_RECORD_REFERENCE)) {
```

Comment:
why can't orchestrator set the bulk scan reference . Services may or may not set but we could always set it even they have set we can override.

Reply:
We discussed this recently. At the moment this is delegated to the client service.